staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -114,6 +114,40 @@ type RollingUpdateStatefulSetStrategy struct {
 	Partition *int32 `json:"partition,omitempty" protobuf:"varint,1,opt,name=partition"`
 }
 
+// PersistentVolumeClaimDeletePolicyType is a string enumeration of the policies that will determine
+// when volumes from the VolumeClaimTemplates will be deleted when the controlling StatefulSet is
+// deleted or scaled down.
+type PersistentVolumeClaimDeletePolicyType string
+
+const (
+	// RetainPersistentVolumeClaimDeletePolicyType is the default
+	// PersistentVolumeClaimDeletePolicy and specifies that
+	// PersistentVolumeClaims associated with StatefulSet VolumeClaimTemplates
+	// will not be deleted.
+	RetainPersistentVolumeClaimDeletePolicyType PersistentVolumeClaimDeletePolicyType = "Retain"
+	// DeletePersistentVolumeClaimDeletePolicyType specifies that
+	// PersistentVolumeClaims associated with StatefulSet VolumeClaimTemplates
+	// will be deleted in the scenario specified in
+	// StatefulSetPersistentVolumeClaimPolicy.
+	DeletePersistentVolumeClaimDeletePolicyType PersistentVolumeClaimDeletePolicyType = "Delete"
+)
+
+// StatefulSetPersistentVolumeClaimPolicy describes the policy used for PVCs
+// created from the StatefulSet VolumeClaimTemplates.
+type StatefulSetPersistentVolumeClaimPolicy struct {
+	// OnSetDeletion specifies what happens to PVCs created from StatefulSet
+	// VolumeClaimTemplates when the StatefulSet is deleted. The default policy
+	// of `Retain` causes PVCs to not be affected by StatefulSet deletion. The
+	// `Delete` policy causes those PVCs to be deleted.
+	OnSetDeletion PersistentVolumeClaimDeletePolicyType `json:"onSetDeletion,omitempty" protobuf:"bytes,1,opt,name=onSetDeletion,casttype=PersistentVolumeClaimDeletePolicyType"`
+	// OnScaleDown specifies what happens to PVCs created from StatefulSet
+	// VolumeClaimTemplates when the StatefulSet is scaled down. The default
+	// policy of `Retain` causes PVCs to not be affected by a scaledown. The
+	// `Delete` policy causes the associated PVCs for condemned pods to be
```

Comment:
`condemned` is unusual verbiage and I would probably not use that phrasing for a variety of reasons.  Suggest instead:

> The `Delete` policy causes the associated PVCs for the excess pods (above the replica count) to be deleted